### ISSUE TITLE
[quick search] Improve quick search

### ISSUE THREAD

**Issue by antonylebechec** [labsquare/cutevariant#375]
Proposition to improve quick search:
- [ ] Quick search bar error in focus (need to unfocus main window to focus on quick search bar)
- [ ] Apply quick filter to the current selection of variants
- [ ] Configure (in config.yml) "gene" field (default "ann.gene")
- [x] Allow position define for 1 position (pattern "CHR:POS")
- [ ] Configure (in config.yml) multiple fields to search into (not only "gene", but also "transcript", "clinvar"...)
- [ ] Allow applying quick search to existing filter (add to filters with AND operator). By default, with a checkbox to apply to all (source)
- [ ] Allow multiple search with pattern 
   - like NCBI query language
      - Example in NCBI: Human[Organism] AND BRCA*[Gene Name]
      - Example in CuteVariant: BRCA[ann.gene] AND NM_007294[ann.transcript]
   - in VQL
      - Example: ann.gene LIKE '%BRCA%' AND ann.transcript LIKE '%NM_007294%'

**Comment by antonylebechec** [comment]
Some fix and new features in "branch fix_gene_view":
- quick search added to existing filters
- quick search "removed" from existing filters (if not changed)
- search on one position (pattern "CHR:POS")
- search on annotations (ann.*)
- search on all other fields
- including operators (=|>|>=|<|<=|!=|=~|!~|IN|!IN|HAS|!HAS)
- detecting IN operator with a list of values with comma "," 
- auto detection of pattern (symbol=*RAS -> symbol =~ RAS) to be user friendly

![quicksearch](https://user-images.githubusercontent.com/40463532/162845236-e3e58e4a-4a1f-4d01-9dc7-153dd2dee9ed.gif)
![quicksearch2](https://user-images.githubusercontent.com/40463532/162845242-dbf9b365-a825-44ce-afcc-d99f4ad084c5.gif)
![quicksearch3](https://user-images.githubusercontent.com/40463532/162845250-efb573c2-2fc0-4677-83ce-7d020bd19fdd.gif)
![quicksearch4](https://user-images.githubusercontent.com/40463532/162845256-7278eb08-8a4a-43df-b912-9a02a239a5bd.gif)

**Comment by dridk** [comment]
- Create different pattern : 
-- Gene Name
-- Chr3:pos-end
-- VQL

**Comment by dridk** [comment]
- Create different pattern : 
-- Gene Name
-- Chr3:pos-end
-- VQL

**Comment by dridk** [comment]
from the new branch #375 , @SteampunkIslande  disable existing filter extension . Because, searching two gene name will generate a filter like "gene='CFTR' AND gene='GJB2'". 
I suggest to extend existing filter with quick filter only for VQL statement.

**Comment by antonylebechec** [comment]
You're right. VQL is the answer to select multiple genes, such as "ann.gene IN KRAS,NRAS"

**Comment by dridk** [comment]
je clos l'issue pour la release.. On rouvrira pour améliorer

**Comment by antonylebechec** [comment]
On ne peut pas la clore tant que :
- la requête VQL construite (par quick search) n'est pas ajoutée aux "filters" existants
- le quick search à "vide" n'enlève pas la requête VQL construite (par quick search)

On peut le faire ne mémorisant la requête construite, puis l'enlever soit 
1. quand le requête quick search est "vide",
2. quand une autre requête est demandée
